mono/metadata/icall.c

```diff
@@ -7928,9 +7930,12 @@ type_array_from_modifiers (MonoImage *image, MonoType *type, int optional, MonoE
 	res = mono_array_new_handle (domain, mono_defaults.systemtype_class, count, error);
 	goto_if_nok (error, fail);
 	count = 0;
-	for (i = cmods->count - 1; i >= 0; --i) {
-		if ((optional && !cmods->modifiers [i].required) || (!optional && cmods->modifiers [i].required)) {
-			if (!add_modifier_to_array (domain, image, &cmods->modifiers [i], res, count , error))
+	for (i = cmod_count - 1; i >= 0; --i) {
+		gboolean required;
```

Comment:
How does this code get hit/tested? Might be good to document. It seems difficult to go from the type of breakage you'd see if this misbehaved back to this code.